File: StokryptoAndroid/app/src/main/java/app/web/bachbntdev/stokrypto/data/models/AssetModel.kt
package app.web.bachbntdev.stokrypto.data.models

import app.web.bachbntdev.stokrypto.domain.entities.Asset

data class AssetModel(
    override val id: String,
    override val symbol: String,
    override val name: String,
    override val price: Double,
    override val volatility: Double,
    override val volatilityPercent: Double,
    override val volume: Double,
    override val type: String,
    override val unit: String
) : Asset

File: StokryptoIOS/Stokrypto/Data/Models/AssetModel.swift
//
//  AssetModel.swift
//  Stokrypto
//
//  Created by bach.bui on 25/07/2023.
//

import Foundation

class AssetModel: Asset {
    var id: String
    var symbol: String
    var name: String
    var price: Double
    var volatility: Double
    var volatilityPercent: Double
    var volume: Double
    var type: String
    var unit: String
    
    init(id: String, symbol: String, name: String, price: Double, volatility: Double = 0.0, volatilityPercent: Double = 0.0, volume: Double = 0.0, type: String, unit: String) {
        self.id = id
        self.symbol = symbol
        self.name = name
        self.price = price
        self.volatility = volatility
        self.volatilityPercent = volatilityPercent
        self.volume = volume
        self.type = type
        self.unit = unit
    }
}
